### PR TITLE
CORCI-978 experiment, do not merge

### PR DESCRIPTION
**Pull request by marcelarosalesj**
In this experiment I erase content from memcheck-cart.supp and
memcheck-daos-client.supp because I want reports with leaks to get
to valgrindPublish section in the CI.

Signed-off-by: Marcela Rosales <marcela.a.rosales.jimenez@intel.com>

### CODE REVIEW

**Comment by brianjmurrell** [comment]
Do the [errors](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3395/2/valgrindResult/) correlate with the removed suppressions?

I still really have no idea how to read those valgrind reports.

Maybe @ashleypittman can provide an opinion.

**Comment by ashleypittman** [comment]
There seem to be mostly four errors that are reported, three relating to dl_open() which seems to be prone to leaks and is only used at startup, and one uninitialised param error which would be best resolved.  This is exactly the kind of errors that I'd expect to be suppressed by these files so yes it's what I'd expect.  It seems that most suppressions could simply be removed based on the testing that we're currently doing although some of them might date from when CaRT was doing valgrind testing and covering different code-paths.

NLT is currently configured to fail if valgrind errors are found but this may not be ideal (it was added when we didn't have any Jenkins representation of results), to disable this remove the --exit-errorcode=42 from this line
https://github.com/daos-stack/daos/blob/master/utils/node_local_test.py#L426

I also note there is a third, much larger, suppressions file at utils/test_memcheck.supp containing a number of issues which would ideally also be resolved.

File: src/cart/utils/memcheck-cart.supp [hunk 1]
```diff
@@ -1,169 +0,0 @@
...
-{
-    CRYPTO leaks calloc
-    Memcheck:Leak
-    match-leak-kinds: reachable
-    fun:calloc
-    ...
-    obj:*/libcrypto.so.1.*
-    ...
-}
-{
```
Comment (ashleypittman) [inline src/cart/utils/memcheck-cart.supp:108]: This one is still used.

File: src/cart/utils/memcheck-cart.supp [hunk 2]
```diff
@@ -1,169 +0,0 @@
...
-    FI leak 0
-    Memcheck:Leak
-    match-leak-kinds: reachable
-    fun:calloc
-    fun:_dlerror_run
-    fun:dlopen@@GLIBC_2.2.5
-    ...
-    fun:crt_hg_init
-    fun:crt_init_opt
-}
```
Comment (ashleypittman) [inline src/cart/utils/memcheck-cart.supp:18]: This one is still used.

File: src/cart/utils/memcheck-cart.supp [hunk 3]
```diff
@@ -1,169 +0,0 @@
-{
-    dlopen leaks
-    Memcheck:Leak
-    match-leak-kinds: reachable
-    ...
-    fun:_dl_open
-}
```
Comment (ashleypittman) [inline src/cart/utils/memcheck-cart.supp:7]: This one is also still used.

File: utils/memcheck-daos-client.supp [hunk 4]
```diff
@@ -1,42 +0,0 @@
...
-   ...
-   fun:send
-   fun:sock_comm_send_socket.isra.3
-   ...
-   fun:sock_pe_progress_tx_entry
-   fun:sock_pe_progress_tx_ctx
-   fun:sock_pe_progress_thread
-   fun:start_thread
-   fun:clone
-}
```
Comment (ashleypittman) [inline utils/memcheck-daos-client.supp:14]: And this one is still used, which is the last of the four that I can see reported.

**Pull request closed without merging**